bin/check-dns.rb

```diff
@@ -132,6 +133,19 @@ def check_against_regex(entries, regex)
     critical "Resolved #{config[:domain]} #{config[:type]} did not match #{regex}"
   end
 
+  def check_ips(entries)
+    ips = []
+    entries.answer.rrsets(config[:type]).each do |ip|
+      ips << ip.to_s.split("\t").last.tr("\n", '')
+    end
+    result = (IPAddr.new config[:result]).to_s
+    if ips.any? { |ip| ip.casecmp(result.downcase).zero? }
```

Comment:
`IPAddr.==` actually works on Strings too, thanks to `IPAddr` coercing Strings into `IPAddr` objects when they're passed in, so `result == ip` would have worked.

It does have the unfortunate property that swapping the arguments around *doesn't* work, i.e. `ip == result` would not have worked, so I can see why you'd want to make it explicit by wrapping each IP in `IPAddr.new`.